parcels/interaction/interactionkernelsoa.py

```diff
@@ -0,0 +1,179 @@
+                    logger.warning_once("Some InteractionKernel was not completed succesfully.")
+
+            for particle_idx in active_idx:
+                p = pset[particle_idx]
+                try:
+                    for mutator_func, args in mutator[p.id]:
+                        mutator_func(p, *args)
+                except KeyError:
+                    pass
+
+    def execute(self, pset, endtime, dt, recovery=None, output_file=None, execute_once=False):
+        """Execute this Kernel over a ParticleSet for several timesteps"""
+        pset.collection.state[:] = StateCode.Evaluate
+
+        if abs(dt) < 1e-6 and not execute_once:
+            logger.warning_once("'dt' is too small, causing numerical accuracy limit problems. Please chose a higher 'dt' and rather scale the 'time' axis of the field accordingly. (related issue #762)")
+
+        if recovery is None:
+            recovery = {}
+        elif ErrorCode.ErrorOutOfBounds in recovery and ErrorCode.ErrorThroughSurface not in recovery:
+            recovery[ErrorCode.ErrorThroughSurface] = recovery[ErrorCode.ErrorOutOfBounds]
+        recovery_map = recovery_base_map.copy()
+        recovery_map.update(recovery)
+
+        if pset.fieldset is not None:
+            for g in pset.fieldset.gridset.grids:
+                if len(g.load_chunk) > g.chunk_not_loaded:  # not the case if a field in not called in the kernel
+                    g.load_chunk = np.where(g.load_chunk == g.chunk_loaded_touched,
+                                            g.chunk_deprecated, g.load_chunk)
+
+        # Execute the kernel over the particle set
+        if self.ptype.uses_jit:
+            self.execute_jit(pset, endtime, dt)
+        else:
+            self.execute_python(pset, endtime, dt)
+
+        # Remove all particles that signalled deletion
+        self.remove_deleted(pset, output_file=output_file, endtime=endtime)   # Generalizable version!
+
+        # Identify particles that threw errors
+        n_error = pset.num_error_particles
+
+        while n_error > 0:
+            error_pset = pset.error_particles
+            # Apply recovery kernel
+            for p in error_pset:
+                if p.state == OperationCode.StopExecution:
+                    return
+                if p.state == OperationCode.Repeat:
+                    p.set_state(StateCode.Evaluate)
+                elif p.state == OperationCode.Delete:
+                    pass
+                elif p.state in recovery_map:
+                    recovery_kernel = recovery_map[p.state]
+                    p.set_state(StateCode.Success)
+                    recovery_kernel(p, self.fieldset, p.time)
+                    if p.state == StateCode.Success:
+                        p.set_state(StateCode.Evaluate)
+                else:
+                    logger.warning_once('Deleting particle {} because of non-recoverable error'.format(p.id))
+                    p.delete()
```

Comment:
@erikvansebille : made a similar comment atop when defining the recovery kernels. Question is: do you really want to remove it if it's re-introduced just a few interaction-related PR's later, to facilitate error resolution ? Because when doing so,  this part seems a good starting point in picking up what needs to be handled.

Reply:
Again: exactly right. If we decide to only add error recovery for InteractionKernels in a later extension, and keep the absence of it as a limitation of this first implementation, I would probably prefer to take it out for now.